pkg/api/validate.go

```diff
@@ -222,13 +282,28 @@ func validateAgentPoolProfile(app *AgentPoolProfile) (errs []error) {
 func validateFQDN(p *Properties) (errs []error) {
 	if p == nil {
 		errs = append(errs, fmt.Errorf("masterProfile cannot be nil"))
+		return
 	}
 	if p.FQDN == "" || !isValidHostname(p.FQDN) || !isAzureZone(p.FQDN) {
 		errs = append(errs, fmt.Errorf("invalid properties.fqdn %q", p.FQDN))
 	}
 	return
 }
 
+func validateNetworkProfile(np *NetworkProfile) (errs []error) {
+	if np == nil {
+		errs = append(errs, fmt.Errorf("networkProfile cannot be nil"))
+		return
+	}
```

Comment:
Should this be required in a non custom VNET case?

Reply:
yes, because it contains the vnetCidr.